File: 백준/Gold/3020. 개똥벌레/개똥벌레.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int H = sc.nextInt();

        int[] arr = new int[H];
        for (int i = 0; i < N; i += 2) {
            arr[0] += 1;
            arr[sc.nextInt()] -= 1;
            arr[H - sc.nextInt()] += 1;
        }

        for (int i = 1; i < H; i++) {
            arr[i] += arr[i - 1];
        }

        int minimum = Integer.MAX_VALUE;
        int count = 0;
        for (int num : arr) {
            if (num < minimum) {
                minimum = num;
                count = 1;
            } else if (num == minimum) {
                count++;
            }
        }

        System.out.println(minimum + " " + count);
    }
}

File: 백준/Gold/3020. 개똥벌레/개똥벌레.py
from sys import stdin
input = stdin.readline

N, H = map(int, input().split())

arr = [0] * H
for i in range(0, N, 2):
    arr[0] += 1
    arr[int(input())] -= 1
    arr[H - int(input())] += 1

for i in range(1, H):
    arr[i] += arr[i - 1]

minimum = min(arr)
count = arr.count(minimum)

print(minimum, count)